core/src/main/scala/org/apache/spark/shuffle/BlockStoreShuffleReader.scala

```diff
@@ -47,8 +47,7 @@ private[spark] class BlockStoreShuffleReader[K, C](
       context,
       blockManager.blockStoreClient,
       blockManager,
-      mapOutputTracker.getMapSizesByExecutorId(handle.shuffleId, startPartition, endPartition,
-        SparkEnv.get.conf.get(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL)),
+      mapOutputTracker.getMapSizesByExecutorId(handle.shuffleId, startPartition, endPartition),
```

Comment:
This is the shuffle read side and we need to know the value of `SHUFFLE_USE_OLD_FETCH_PROTOCOL`. I think the bug is in the shuffle write side which is fixed in this PR. Do we really need to change the shuffle read side?

Reply:
This redundant code, since ShuffleWrite writes the mapId based on the `spark.shuffle.useOldFetchProtocol` flag, `MapStatus.mapTaskId` always gives the mapId which is set by the ShuffleWriter